Documentation: Email notification for cron job

katekimani opened the pull request:
Trello card - https://trello.com/c/kjO60kDB/2082-bohemia-back-end-email-notification-script

joebrew reviewed (changes requested):
@katekimani All looking good, but I can't quite get this to work. Here is what happens on stop 10/11:

```
ubuntu@ip-10-0-0-134:~$ echo "Confirm email works test" | mail -s "Cron email test" wanjiru@databrew.ccubuntu@ip-10-0-0-134:~$ tail -10 /var/log/mail.log
Nov 23 11:50:21 ip-10-0-0-134 postfix/smtp[4709]: EBF0D65A46D: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=444, delays=442/0.02/0.92/1.3, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132221 h1sm7862139ilj.8 - gsmtp)
Nov 23 11:50:21 ip-10-0-0-134 postfix/qmgr[4026]: EBF0D65A46D: removed
Nov 23 11:50:21 ip-10-0-0-134 postfix/smtp[4707]: 683BC65A467: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=993, delays=991/0.01/0.94/1.6, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132221 u16sm7404669ilj.6 - gsmtp)
Nov 23 11:50:21 ip-10-0-0-134 postfix/qmgr[4026]: 683BC65A467: removed
Nov 23 11:50:23 ip-10-0-0-134 postfix/pickup[4025]: 94AFF65A46A: uid=1000 from=<ubuntu@ip-10-0-0-134.us-east-2.compute.internal>
Nov 23 11:50:23 ip-10-0-0-134 postfix/cleanup[4714]: 94AFF65A46A: message-id=<20201123115023.94AFF65A46A@bohemia.systems>
Nov 23 11:50:23 ip-10-0-0-134 postfix/qmgr[4026]: 94AFF65A46A: from=<ubuntu@ip-10-0-0-134.us-east-2.compute.internal>, size=423, nrcpt=1 (queue active)
Nov 23 11:50:23 ip-10-0-0-134 postfix/smtp[4709]: connect to smtp.gmail.com[2607:f8b0:4023:1402::6c]:587: Network is unreachable
Nov 23 11:50:25 ip-10-0-0-134 postfix/smtp[4709]: 94AFF65A46A: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=1.6, delays=0.01/0/0.67/0.92, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132225 l9sm7882437ilt.19 - gsmtp)
Nov 23 11:50:25 ip-10-0-0-134 postfix/qmgr[4026]: 94AFF65A46A: removed

```

katekimani commented:
> @katekimani All looking good, but I can't quite get this to work. Here is what happens on stop 10/11:
> 
> ```
> ubuntu@ip-10-0-0-134:~$ echo "Confirm email works test" | mail -s "Cron email test" wanjiru@databrew.ccubuntu@ip-10-0-0-134:~$ tail -10 /var/log/mail.log
> Nov 23 11:50:21 ip-10-0-0-134 postfix/smtp[4709]: EBF0D65A46D: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=444, delays=442/0.02/0.92/1.3, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132221 h1sm7862139ilj.8 - gsmtp)
> Nov 23 11:50:21 ip-10-0-0-134 postfix/qmgr[4026]: EBF0D65A46D: removed
> Nov 23 11:50:21 ip-10-0-0-134 postfix/smtp[4707]: 683BC65A467: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=993, delays=991/0.01/0.94/1.6, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132221 u16sm7404669ilj.6 - gsmtp)
> Nov 23 11:50:21 ip-10-0-0-134 postfix/qmgr[4026]: 683BC65A467: removed
> Nov 23 11:50:23 ip-10-0-0-134 postfix/pickup[4025]: 94AFF65A46A: uid=1000 from=<ubuntu@ip-10-0-0-134.us-east-2.compute.internal>
> Nov 23 11:50:23 ip-10-0-0-134 postfix/cleanup[4714]: 94AFF65A46A: message-id=<20201123115023.94AFF65A46A@bohemia.systems>
> Nov 23 11:50:23 ip-10-0-0-134 postfix/qmgr[4026]: 94AFF65A46A: from=<ubuntu@ip-10-0-0-134.us-east-2.compute.internal>, size=423, nrcpt=1 (queue active)
> Nov 23 11:50:23 ip-10-0-0-134 postfix/smtp[4709]: connect to smtp.gmail.com[2607:f8b0:4023:1402::6c]:587: Network is unreachable
> Nov 23 11:50:25 ip-10-0-0-134 postfix/smtp[4709]: 94AFF65A46A: to=<wanjiru@databrew.cc>, relay=smtp.gmail.com[142.250.112.109]:587, delay=1.6, delays=0.01/0/0.67/0.92, dsn=2.0.0, status=sent (250 2.0.0 OK  1606132225 l9sm7882437ilt.19 - gsmtp)
> Nov 23 11:50:25 ip-10-0-0-134 postfix/qmgr[4026]: 94AFF65A46A: removed
> ```

That last log entry shows it succeeded. ` status=sent (250 2.0.0 OK  1606132225 l9sm7882437ilt.19 - gsmtp)` 
I did receive the emails too. You could test again using `joe@databrew.cc` to get the emails

katekimani commented on guides/data_management_and_analytic_plan.md:
````diff
@@ -1141,6 +1141,72 @@ clean tables), run: `clean_db()`
+
+6.  Add/Update the entries to be:
+    ```
+      myhostname = bohemia.systems
````
```suggestion
      myhostname = bohemia.team
```

katekimani commented on guides/data_management_and_analytic_plan.md:
````diff
@@ -1141,6 +1141,72 @@ clean tables), run: `clean_db()`
+    sudo chmod 750 /etc/postfix/sasl
+    sudo chmod 640 /etc/postfix/sasl/sasl_passwd
+    ```
+
````
```suggestion
10. Restart the postfix service by running `sudo service postfix restart`
```
